snapcraft/_options.py

```diff
@@ -242,7 +243,16 @@ def _set_machine(self, target_deb_arch):
             self.__target_machine = _find_machine(target_deb_arch)
             logger.info('Setting target machine to {!r}'.format(
                 target_deb_arch))
-        self.__machine_info = _ARCH_TRANSLATIONS[self.__target_machine]
+        if sys.platform == 'linux':
+            self.__machine_info = _ARCH_TRANSLATIONS[self.__target_machine]
+        else:
+            self.__machine_info = {
+                'kernel': 'NOBUILD',
+                'deb': 'NOBUILD',
+                'uts_machine': 'NOBUILD',
+                'triplet': 'NOBUILD',
+                'core-dynamic-linker': 'NOBUILD',
+            }
```

Comment:
With the placeholders it looks a little cryptic... can you add a comment here to clarify what the code does?

Maybe `# Provider fake machine info to allow container and store commands to work on non-Linux platforms`.